packages/@aws-cdk/aws-cognito/README.md

```diff
@@ -753,3 +766,374 @@ Existing domains can be imported into CDK apps using `UserPoolDomain.fromDomainN
+
+Permissions can be restricted to specific resources using the `grantResource` methods. These methods are similar to  
+the grant methods except that the the first argument they tale is a `resourceArn` for the resource the permissions  
+should pertain to:
+
+```ts
+// Grant permissions to authenticated users
+identityPool.grantUserResource('arn:aws:execute-api:us-east-1:*:my-user-api/prod', 'execute-api:*');
+
+// Grant permissions to unauthenticated guest users
+identityPool.grantGuestResource('arn:aws:execute-api:us-east-1:*:my-guest-api/prod', 'execute-api:*');
+```
+
+The default authenticated and unauthenticated roles are preconfigured to allow users to assume their respective  
+credentials using the `sts:AssumeRoleWithWebIdentity` action. A different action can be used to assume the role using 
+the `assumeAction` field:
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  assumeAction: 'sts:AssumeRole',
+  grantUserActions: ['execute-api'],
+});
+```
+
+### Authentication Providers
+
+Authenticated identities belong to users who are authenticated by a public login provider (Amazon Cognito user pools,  
+Login with Amazon, Sign in with Apple, Facebook, Google, SAML, or any OpenID Connect Providers) or a developer provider  
+(your own backend authentication process).
+
+[Authentication providers](https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html) can be associated with an Identity Pool by first associating them with a Cognito User Pool or by  
+associating the provider directly with the identity pool.
+
+#### User Pool Authentication Provider
+
+In order to attach an existing user pool to an identity pool, several elements not present in the `UserPool` construct  
+are needed, so the `userPool` field of `authenticationProviders` requires an object or class that implements the  
+`IUserPoolAuthenticationProvider` interface. The most common implementation of this interface is the `UserPoolAuthenticationProvider`  
+class. The `UserPoolAuthenticationProvider` class is a special class that configures a `UserPool` and a `UserPoolClient` for use as an authentication provider. 
+
+```ts
+const userPool = new cognito.UserPool(this, 'Pool');
+
+// using the UserPoolAuthenticationProvider class
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  grantUserActions: ['execute-api'],
+  authenticationProviders: {
+    userPool: new UserPoolAuthenticationProvider({ userPool }),
+  },
+});
+
+// using an object that implements the `IUserPoolAuthenticationProvider` interface
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  grantUserActions: ['execute-api'],
+  authenticationProviders: {
+    userPool: {
+      // The User Pool Client's clientId
+      clientId: 'my-user-pool-client',
+      // A list of IUserPoolIdentityProviders
+      identityProviders: [...identityProviders],
```

Comment:
`identityProviders` is a public property of the `UserPool` construct. `The CloudFormation` template needs that list of `identityProviders` from the `UserPool` and the `clientId` from the `UserPoolClient` to connect it to the `IdentityPool`. The `UserPoolAuthenticationProvider` class does this under the hood, but this is just showing it can also be done manually.